gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Client.java

```diff
@@ -518,9 +548,23 @@ protected void initializeImplementation() {
 
         @Override
         public CompletableFuture<ResultSet> submitAsync(final Bytecode bytecode) {
+            return submitAsync(bytecode, RequestOptions.EMPTY);
+        }
+
+        @Override
+        public CompletableFuture<ResultSet> submitAsync(final Bytecode bytecode, final RequestOptions options) {
             try {
-                return submitAsync(buildMessage(RequestMessage.build(Tokens.OPS_BYTECODE)
-                        .processor("traversal").addArg(Tokens.ARGS_GREMLIN, bytecode)).create());
+                // need to call buildMessage() right away to get client specific configurations, that way request specific
+                // ones can override as needed
+                final RequestMessage.Builder request = buildMessage(RequestMessage.build(Tokens.OPS_BYTECODE)
+                                                                                  .processor("traversal")
+                                                                                  .addArg(Tokens.ARGS_GREMLIN, bytecode));
+
+                // apply settings if they were made available
+                options.getBatchSize().ifPresent(batchSize -> request.add(Tokens.ARGS_BATCH_SIZE, batchSize));
+                options.getTimeout().ifPresent(timeout -> request.add(Tokens.ARGS_SCRIPT_EVAL_TIMEOUT, timeout));
```

Comment:
If `ARGS_SCRIPT_EVAL_TIMEOUT` now configures bytecode queries, should it change to something without "script"? Maybe deprecate this one for now and introduce another one and eventually remove the deprecated one

Reply:
i've thought of that before. lots of changes but maybe that should happen........created this: https://issues.apache.org/jira/browse/TINKERPOP-2213 to be considered as part of a separate body of work.